language/src/modules/Language.rb

```diff
@@ -915,53 +916,20 @@ def Selection
 
     # Save state to target.
     def Save
-      loc = locale_to_install
-
-      @localed_conf = {} if @localed_conf.nil?
-
-      if Builtins.find(loc, "zh_HK") == 0
-        @localed_conf["LC_MESSAGES"] = "zh_TW"
-      elsif @localed_conf["LC_MESSAGES"] == "zh_TW"
-        # FIXME ugly hack: see bug #47711
-        @localed_conf.delete("LC_MESSAGES")
-      end
-
       SCR.Write(path(".sysconfig.language.INSTALLED_LANGUAGES"), @languages)
       SCR.Write(path(".sysconfig.language"), nil)
 
-      @localed_conf["LANG"] = loc
-      log.info("Locale: #{@localed_conf}")
-      locale_out1 = @localed_conf.map do | key, val |
-        "#{key}=#{val}"
-      end
-      log.info("Locale: #{locale_out1}")
-      locale_out=locale_out1.join(",")
-      log.info("Locale: #{locale_out}")
+      command = save_command_for(locale)
 
-      cmd = if Stage.initial
-        # do use --root option, running in chroot does not work
-        "/usr/bin/systemd-firstboot --root #{Installation.destdir.shellescape} --locale #{loc.shellescape}"
-      else
-        # this sets both the locale (see "man localectl")
-        "/usr/bin/localectl set-locale #{locale_out.shellescape}"
-      end
-      log.info "Making language setting persistent: #{cmd}"
-      result = if Stage.initial
-        WFM.Execute(path(".local.bash_output"), cmd)
-      else
-        SCR.Execute(path(".target.bash_output"), cmd)
-      end
-      if result["exit"] != 0
-        log.error "Language configuration not written. Failed to execute '#{cmd}'"
-        log.error "output: #{result.inspect}"
-        # TRANSLATORS: the "%s" is replaced by the executed command
-        Report.Error(_("Could not save the language setting, the command\n%s\nfailed.") % cmd)
-      else
-        log.info "output: #{result.inspect}"
-      end
+      log.info("Making language settings persistent: #{command}")
 
-      Builtins.y2milestone("Saved data for language: <%1>", loc)
+      Yast::Execute.locally!(command.split)
```

Comment:
sorry, this cannot work this way. Using split is very bad idea. We shellescape arguments, but it can be something like `British English` which we escape to `British\ English` and this split break it. Also Cheetah does not use shell. So shellescape here will break it heavily.